src/api/session/SessionManager.ts

```diff
@@ -167,7 +176,22 @@ export class SessionManager implements ISessionManager {
     await this.workspaceSwitch({ targetWorkspace, workspaceCreation })
   }
 
-  public logout() {
+  public async logout(logoutOptions?: LogoutOptions) {
+    const { invalidateBrowserAuthCookie }: LogoutOptions = { invalidateBrowserAuthCookie: false, ...logoutOptions }
+    if (this.token) {
+      const vtexId = VTEXID.createClient()
+      try {
+        await vtexId.invalidateToolbeltToken(this.token)
+      } catch (err) {
+        ErrorReport.createAndMaybeRegisterOnTelemetry({ originalError: err })
+        logger.error('Unable to invalidate local token')
+      }
+    }
+
+    if (invalidateBrowserAuthCookie && this.account) {
+      VTEXID.invalidateBrowserAuthCookie(this.account)
+    }
```

Comment:
If the first invalidation fails, shouldn't you fail the whole action? I mean, the invalidation of the local token might fail and you'd still open the browser and log out the user there.